### ISSUE TITLE
1.54.1 update today broke the VSCode JavaScript debugger on my machine.  (Windows10x64)

### ISSUE THREAD

**Issue by dirck** [microsoft/vscode#118679]
Issue Type: <b>Bug</b>

After allowing VSCode to run the 1.54.1 update today, when attempting to debug javascript, vscode presents:

"Couldn't find a debug adapter descriptor for debug type 'pwa-node' (extension may have failed to activate)"

Debug efforts:

* Uninstalled vscode
* Reinstalled vscode (Windows "System Installer" x64)

no change

* Uninstalled vscode
* Completely hosed out %APPDATA%/code
* Reinstalled vscode (Windows "System Installer" x64)

no change

(note the previously installed extensions still appear in vscode after hosing out %APPDATA%/code, not sure where they live.)

Tested with and without --disable-extensions option
___

Highlights from `code --disable-extensions --verbose --log debug`

Multiple occurrences of:
```
[32776:0310/160830.059:INFO:CONSOLE(25)] "%cTRACE color: #888 Error: ENOENT: no such file or directory, open 'c:\Program Files\Microsoft VS Code\resources\app\extensions\node_modules\package.json'", source: file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js (25)
```

One time:
```
[32776:0310/160830.788:INFO:CONSOLE(626)] "%cTRACE color: #888 CommandService#executeCommand extension.js-debug.clearAutoAttachVariables", source: file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js (626)
```

Previous log before second reinstall includes both of the above plus this message, not present in the latest log:
```
[34784:0310/155307.979:INFO:CONSOLE(1857)] "Activating extension 'ms-vscode.js-debug' failed: a DebugAdapterDescriptorFactory can only be registered from the extension that defines the 'pwa-extensionHost' debugger..", source: file:///C:/PROGRA~1/MIFA7F~1/resources/app/out/vs/workbench/workbench.desktop.main.js (1857)
```

VS Code version: Code 1.54.1 (f30a9b73e8ffc278e71575118b6bf568f04587c8, 2021-03-04T22:38:31.419Z)
OS version: Windows_NT x64 10.0.19042

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) CPU E3-1270 V2 @ 3.50GHz (8 x 3492)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.95GB (18.63GB free)|
|Process Argv|--disable-extensions --verbose --log debug --crash-reporter-id 39bea185-078d-4173-a244-da107d90ee57|
|Screen Reader|no|
|VM|0%|
</details>Extensions disabled<details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
vstry244cf:30256637
pythonvsdeb440:30248342
pythonvsded773:30248341
pythonvspyt875:30259475
dockersubset:30270708
pythontb:30265425
vspre833cf:30267465

```

</details>

**Comment by dirck** [comment]
After I made the report the debugger started working.  WTF?

**Comment by dirck** [comment]
It came back and I still haven't pinned it to anything.  Might be related to an extension doing something globally, even though it still happens with extensions disabled.

**Comment by dirck** [comment]
I've got something I can reproduce; if I run it like this:
```
"C:\Program Files\Microsoft VS Code\bin\code.CMD" .
```
it works, and if I run it like this:
```
c:\progra~1\MIFA7F~1\bin\code.cmd .
```
it fails.

Weird.

**Comment by connor4312** [comment]
cc Andre:

> Activating extension 'ms-vscode.js-debug' failed: a DebugAdapterDescriptorFactory can only be registered from the extension that defines the 'pwa-extensionHost' debugger..", source: file:///C:/PROGRA~1/MIFA7F~1/resources/app/out/vs/workbench/workbench.desktop.main.js

**Comment by weinand** [comment]
@connor4312 I have never seen the check from above failing randomly.
Here is the code:
https://github.com/microsoft/vscode/blob/d574f8e9163f08213376560e5ab321fd02e48332/src/vs/workbench/api/common/extHostDebugService.ts#L745-L760

Did something change in that area?

**Comment by connor4312** [comment]
I have not changed anything there. But also looking at the "notok.log" the user posted in https://github.com/microsoft/vscode/issues/118679#issuecomment-796389620, it looks like there are quite a few other weird errors, so maybe this is deeper down in the stack.

**Comment by Onurtag** [comment]
Having the same problem.   
#118562 might be similar to this issue.

**Comment by Bellybutn278** [comment]
Have you got this corrected

**Comment by Bellybutn278** [comment]
GCC 10.2.0

**Comment by dirck** [comment]
Verified fixed in 1.54.3

**Comment by weinand** [comment]
@dirck thanks for verifying!
